kubernetes/resource_kubernetes_yaml.go

```diff
@@ -135,7 +135,7 @@ func resourceKubernetesYAMLCreate(d *schema.ResourceData, meta interface{}) erro
 		return fmt.Errorf("failed to create resource in kubernetes: %+v", err)
 	}
 
-	d.SetId(response.GetSelfLink())
+	d.SetId(response.GetName())
```

Comment:
Not 100% sure this is the correct replacement (maybe namespace+name?) but this works in my limited testing